src/Features/CSharp/CodeRefactorings/InlineTemporary/InlineTemporaryCodeRefactoringProvider.cs

```diff
@@ -504,37 +504,40 @@ private static SyntaxNode GetTopMostParentingExpression(ExpressionSyntax express
             var inlinedExprNodes = syntaxRootAfterInline.GetAnnotatedNodesAndTokens(ExpressionToInlineAnnotation);
             Debug.Assert(originalIdentifierNodes.Count() == inlinedExprNodes.Count());
 
-            var originalNodesEnum = originalIdentifierNodes.GetEnumerator();
-            var inlinedNodesEnum = inlinedExprNodes.GetEnumerator();
             Dictionary<SyntaxNode, SyntaxNode> replacementNodesWithChangedSemantics = null;
-
-            while (originalNodesEnum.MoveNext())
+            using (var originalNodesEnum = originalIdentifierNodes.GetEnumerator())
             {
-                inlinedNodesEnum.MoveNext();
-                var originalNode = originalNodesEnum.Current;
+                using (var inlinedNodesOrTokensEnum = inlinedExprNodes.GetEnumerator())
+                {
+                    while (originalNodesEnum.MoveNext())
+                    {
+                        inlinedNodesOrTokensEnum.MoveNext();
+                        var originalNode = originalNodesEnum.Current;
 
-                // expressionToInline is Parenthesized prior to replacement, so get the parenting parenthesized expression.
-                var inlinedNode = (ExpressionSyntax)inlinedNodesEnum.Current.AsNode().Parent;
-                Debug.Assert(inlinedNode.IsKind(SyntaxKind.ParenthesizedExpression));
+                        // expressionToInline is Parenthesized prior to replacement, so get the parenting parenthesized expression.
+                        var inlinedNode = (ExpressionSyntax)inlinedNodesOrTokensEnum.Current.Parent;
+                        Debug.Assert(inlinedNode.IsKind(SyntaxKind.ParenthesizedExpression));
 
-                // inlinedNode is the expanded form of the actual initializer expression in the original document.
-                // We have annotated the inner initializer with a special syntax annotation "InitializerAnnotation".
-                // Get this annotated node and compute the symbol info for this node in the inlined document.
-                var innerInitializerInInlineNode = (ExpressionSyntax)inlinedNode.GetAnnotatedNodesAndTokens(InitializerAnnotation).First().AsNode();
-                var newInializerSymbolInfo = newSemanticModelForInlinedDocument.GetSymbolInfo(innerInitializerInInlineNode, cancellationToken);
+                        // inlinedNode is the expanded form of the actual initializer expression in the original document.
+                        // We have annotated the inner initializer with a special syntax annotation "InitializerAnnotation".
+                        // Get this annotated node and compute the symbol info for this node in the inlined document.
+                        var innerInitializerInInlineNode = (ExpressionSyntax)inlinedNode.GetAnnotatedNodesAndTokens(InitializerAnnotation).First().AsNode();
```

Comment:
@Pilchie Yes, that could be an issue. The place I fixed, we were looking for the parent of the SyntaxNodeorToken, which was a Node. so that fix was ok.

In this case though, I dont what the right behavior should be. Should we throw a contract failure? or get the parent node if the resultant happens to be a token. Without the actual scenarios the correct behavior is tough to speculate about